src/core/mapping/base_mapper.cc

```diff
@@ -682,9 +683,34 @@ bool BaseMapper::map_legate_store(const MapperContext ctx,
   // Generate layout constraints from the store mapping
   LayoutConstraintSet layout_constraints;
   mapping.populate_layout_constraints(layout_constraints);
+  auto& fields = layout_constraints.field_constraint.field_set;
+
+  // We need to hold the instance manager lock as we're about to try to find an instance
+  AutoLock reduction_lock(ctx, reduction_instances->manager_lock());
```

Comment:
can't this be moved inside the if statement at line 695? we shouldn't hold locks unnecessarily.

Reply:
sure, makes sense